components/renku_data_services/users/db.py

```diff
@@ -277,3 +303,99 @@ async def events_sync(self, kc_api: IKeycloakAPI):
                     logging.info(
                         f"Updated the latest sync event timestamp in the database: {current_sync_latest_utc_timestamp}"
                     )
+
+
+class UserSecretRepo:
+    """An adapter for accessing users secrets."""
+
+    def __init__(
+        self,
+        session_maker: Callable[..., AsyncSession],
+    ):
+        self.session_maker = session_maker
+
+    @only_authenticated
+    async def get_secrets(self, requested_by: APIUser, user_id: str) -> list[Secret]:
+        """Get a specific user secret from the database."""
+        if user_id != requested_by.id:
+            raise errors.Unauthorized(message="Cannot list secrets.")
```

Comment:
Since this check is pertinent to all functions, a decorator would make things simpler. It could have the error message as parameter.